### PR TITLE
feat(billing): invoice list

### PR DESCRIPTION
**Pull request by bdebon**
# What does this PR do?

> Link to the JIRA ticket

- Invoices list
- Filtering by year
- Download one
- Download many
- tests

https://user-images.githubusercontent.com/6163954/224091449-ba7fab82-1057-42aa-b76b-87277d0f8d3e.mp4

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [x] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by evoxmusic** [comment]
A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [[PR] staging - feat(billing): invoice list - 2023-03-09T16:40:09Z](https://new.console.qovery.com/organization/141c07c8-0dd9-4623-983b-3fdd61867255/project/977bd069-20f7-4b24-bd95-28b44174cf43/environment/2bf57625-e2e8-4f16-9926-746b499533d5/services/general)

 Another comment will be posted when deployments are terminated

File: libs/pages/settings/src/lib/ui/page-organization-billing-summary/invoices-list/invoices-list.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    <div className="mb-3">
+      <div className="flex items-center justify-between mb-3 py-5 ">
+        <h1 className="h5 text-text-700 mb-2">Invoices</h1>
+        <div className="flex gap-3">
+          <InputSelectSmall
+            dataTestId="year-select"
+            items={props.yearsForSorting || []}
+            name="year"
+            inputClassName="h-9 !py-[0.4rem]"
+            className="w-[200px] relative z-[1000]"
```
Comment (RemiBonnet) [inline libs/pages/settings/src/lib/ui/page-organization-billing-summary/invoices-list/invoices-list.tsx:70]: Why the z-index is super huge? Can you use one of this list ?
Capture d’écran 2023-03-13 à 10 10 18

File: libs/pages/settings/src/lib/ui/page-organization-billing-summary/invoices-list/table-row-invoice/table-row-invoice.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import { Invoice, InvoiceStatusEnum } from 'qovery-typescript-axios'
+// eslint-disable-next-line @nrwl/nx/enforce-module-boundaries
```
Comment (RemiBonnet) [inline libs/pages/settings/src/lib/ui/page-organization-billing-summary/invoices-list/table-row-invoice/table-row-invoice.tsx:2]: Why do we have this one? An oversight?

File: libs/pages/settings/src/lib/ui/page-organization-billing-summary/invoices-list/table-row-invoice/table-row-invoice.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        <div className="px-4 text-xs text-text-500 font-medium">
+          <Tag className={`border  ${statusBadgeClassNames[data.status]}`} size={TagSize.SMALL}>
+            {data.status.replace('_', ' ')}
+          </Tag>
+        </div>
+        <div className="px-4 text-xs text-text-500 font-medium">
+          {costToHuman(data.total_in_cents / 100, data.currency_code)}
+        </div>
+        <div className="px-4 text-xs text-text-500 font-medium">
+          <ButtonIcon
```
Comment (RemiBonnet) [inline libs/pages/settings/src/lib/ui/page-organization-billing-summary/invoices-list/table-row-invoice/table-row-invoice.tsx:58]: Can we have a loading status and a toaster for this request?

File: libs/shared/ui/src/lib/components/tag/tag.tsx [hunk 4]
```diff
@@ -15,11 +16,12 @@ export function Tag(props: TagProps) {
 
   const normalClassName = size === TagSize.NORMAL ? 'h-7 px-2' : ''
   const bigClassName = size === TagSize.BIG ? 'h-8 px-3' : ''
+  const smalllassName = size === TagSize.SMALL ? 'h-5 px-1.5' : ''
 
   return (
     <span
       data-testid={dataTestId}
-      className={`rounded text-xs items-center inline-flex font-bold ${normalClassName} ${bigClassName} ${className}`}
+      className={`rounded text-xs items-center inline-flex font-bold ${normalClassName} ${bigClassName} ${smalllassName} ${className}`}
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/tag/tag.tsx:24]: Typo with `smalllassName`

**Review by RemiBonnet — COMMENTED** [review]
Hey @bdebon well done 💯, once this PR is merged, is it possible to add the promo code modal and a release with this billing? It will allow us to start the migration to console.qovery.com

Here my feedback: 
- Download all seem to have a problem with the API? Doesn't work on my side

- Can you add a loader button on this one ? with a toaster for the action should be great !
Capture d’écran 2023-03-13 à 10 24 07

- The filter by date is weird because we have only one value per line, is it necessary to keep it?
Capture d’écran 2023-03-13 à 10 25 39

- I find it a shame to click on "Filter by year" to reset the filter, can we have an action "All" and use this label as a placeholder?
Capture d’écran 2023-03-13 à 10 26 58

**Pull request merged**